crates/node-builder/src/exex.rs

```diff
@@ -0,0 +1,101 @@
+#![allow(dead_code)]
+
+use crate::FullNodeTypes;
+use futures::{future::BoxFuture, Future, FutureExt, Stream};
+use reth_node_core::{
+    dirs::{ChainPath, DataDirPath},
+    node_config::NodeConfig,
+};
+use reth_primitives::{BlockNumber, Head};
+use reth_tasks::TaskExecutor;
+use std::{borrow::Cow, pin::Pin};
+
+/// An ExEx (Execution Extension) that processes new blocks and emits events on a stream.
+pub trait ExEx: Stream<Item = ExExEvent> + Send + 'static {
```

Comment:
make the item a Result<Event>

or maybe turn this into a future that resolves to Result<(), E>? and have events over channels

Reply:
let's make it a Future instead of a Stream that resolves into Result?